gmso/utils/compatibility.py

```diff
@@ -59,14 +82,24 @@ def check_compatibility(topology, accepted_potentials):
     return potential_forms_dict
 
 
-# @lru_cache()
 def _check_single_potential(potential, accepted_potentials):
     """Check to see if a single given potential is in the list of accepted potentials."""
     ind_var = potential.independent_variables
-    u_dims = {para.units.dimensions for para in potential.parameters.values()}
+    u_dims = [
+        symplify_str_eqn(para.units.dimensions)
+        for para in potential.parameters.values()
+    ]
+    for i in range(len(u_dims)):
+        if "temperature" in u_dims[i]:
+            energy_dimsStr = str(replace_temp_with_energy(u_dims[i]))
+            energy_dimsStr = symplify_str_eqn(energy_dimsStr)
+            u_dims[i] = energy_dimsStr
+    u_dims = set(u_dims)
```

Comment:
I think the set is just for comparison of the reference potential to the testing potential. If I turn it off, it fails immediately, because then the two lists that are compared are not always in the same order. 

> If we worry about the order, we can sort both list before comparing.
This is a viable option. However, per speed testing, potential validation and this function actually are a larger bit of the overall function, something between 30-40% of the overall run time. This function gets called A LOT because we're performing these operations on every potential in the system. So an additional sorting step would cause even more time spent here already.